source/common/upstream/health_checker_impl.cc

```diff
@@ -51,8 +53,8 @@ HealthCheckerFactory::create(const envoy::api::v2::core::HealthCheck& hc_config,
     return std::make_shared<ProdGrpcHealthCheckerImpl>(cluster, hc_config, dispatcher, runtime,
                                                        random);
   default:
-    // TODO(htuch): This should be subsumed eventually by the constraint checking in #1308.
-    throw EnvoyException("Health checker type not set");
+    // Checked by schema.
+    NOT_REACHED;
```

Comment:
mild preference for not doing code changes in move PRs unless necessary.  Fine to leave this one in but a note for future PRs

Reply:
@htuch this and the removal of HealthCheckerFactoryTest.MissingFieldiException LGTY?

Reply:
Yes sorry, I just saw it and did it. I'm overeager. :)